### DISCUSSION TITLE
Removing support for Android 7 and older versions

### DISCUSSION THREAD

**Discussion by naveensingh** [FossifyOrg/General-Discussion#241] [Announcements]
Hello!

Android 7 and older versions will no longer be officially supported. This is to reduce testing/development workload. If you are using a such device, please look for alternatives.

Thank you.

**Comment by ronniedroid** [comment]
Finally. I would say anything below 11 can go.

**Reply by naveensingh** [reply]
You have to be considerate at larger scales :)

**Reply by ronniedroid** [reply]
Oh wow, 5% of users are still on Android 9, I was mistaken then.

**Reply by marc-bouvier** [reply]
Supporting old android versions has a considerable positive impact on devices replacement rate.

It is great news for the planet that you still support android 8+.

Thank you for the testing efforts that is involved. 🙏

**Reply by dp35654** [reply]
My Huawie Android tablet WiFi only is I believe Android 8 haven't used it in months cuz I have to remove the plastic screen & the old glue and the thick screen protection I put on it, it's not broke it's just warped from leaving it on the round hump of a old couch! FYI don't leave a device on a concaved surface, lasted for years Unfortunately due to the Trump ban Google kept removing app's that could be deemed a Security threat like the recorder Huawie file's and a couple of others! Even though purchased 2018/19 it's Google certified and much better then Apple and cheaper in my opinion but it would take away from Apple & Google stocks I'm sure! If I could find a similar tablet that you can install what you want I'd go for it cuz I don't feel like spending an hour on a maybe, Samsung A9 tablet WiFi only was horrific and only allowed Meet and I think whatsup? Super slow. I believe the higher we go with Android the less leeway we have to customize are device but I'm not a developer and defiantly can't afford to buy something new because I was playing around

**Reply by ZarVladimirII** [reply]
I absolutely disagree.
Older phones are often used as second devices, sometimes without a SIM-Card for tasks fossify Apps are made for. Completely offline even.
Nevertheless if you don't open links or stuff from outside and nothing says 'I am here with an old device' you do not need to care about security issues.  I would then have problems since a decade+.

Also people started to use their phones much longer because the hardware is not so fast outdated and thus overstrained since 2019/2020.
Smart people would like to use their electronic devices 10+ years, if the world around them would allow but does not. (mainly because of software)

And personally I don't want android 12+ because of the design saying to me "you are stupid" "you the user need a oversimply idiotic design", I felt offended seeing it the first time and still. Hate when I have to troubleshoot on stock android 12+ devices.
Android 10 was a design breakthrough and necessity, should have keept this for longer.

I am fine when stopping testing on oldest versions while still installable.

**Comment by TecnikOfficial** [comment]
i thought open source software had good old devices combability... coz google unnecessary yearly new android version(thts how they get u) will make u replace perfectly useful phone if app devs stop supporting older android version . for me no problem as [heliboard](https://github.com/Helium314/HeliBoard) still runs even on android 5 devices 
although its totally understandable as u r the sole creator of fossify when is actually a great incentive ... i personally debloated by android and replaced most apps with your apps tysm for your efforts :)

**Reply by aritorhamo** [reply]
I'm sorry to hear that my phone from 2016 will not be supported anymore. It has Android 6.0, and it's impossible to get a later version installed, even with a custom ROM. The phone works quite well, especially after I used Universal Android Debloater to make the system lighter and faster.  The phone even still has the original battery.

I hope I can continue using version 1.20 of the phone app. I just discovered Fossify, and had already replaced a few apps by Simple Mobile Tools with your apps when I noticed this announcement. I wonder whether ending support for old phones concerns all Fossify apps, or only Phone.

**Reply by Aga-C** [reply]
@aritorhamo It's for all Fossify apps. We are dropping Android 7 (and older) support with the next update of each app.

**Reply by aritorhamo** [reply]
To clarify, can I still use the older versions of the apps? I have installed the Fossify apps I already have on my phone from F-Droid. If I can continue to use the older versions, do you know if F-Droid will be aware of the situation and not suggest later, non-functioning versions for me (and continue having the older version available)? Thanks!

**Reply by Aga-C** [reply]
@aritorhamo You can still use them. Also, F-Droid can check if the app will work on your device - just be sure that you have `Include incompatible versions` disabled in Settings. Or you can set to ignore updates for the particular app.

**Reply by aritorhamo** [reply]
That's nice to hear. Perhaps I'll get full ten years of use out of my phone then :-).

**Comment by pandaroux007** [comment]
Good morning,
I just discovered Fossify and I am very disappointed to learn that my old Samsung Galaxy J5 of 2016 will not have non-Samsung tools! In particular, I wanted to replace the Samsung messaging with `fossify messages' but as the support stops at Android 8 and that I am on Android 7 (still widely used on the many old phones in circulation) this is not possible. ..

Can the support stop at version 7 of Android, to support almost all of the Android phones currently in operation? It will be really great (even if I hear that it represents a lot of work, but I do not realize how much, so if it is really impossible it does not matter your development work remains incredible and very appreciated! ))

(*I use google translation to write, sorry for the sometimes bizarre formulations* :woozy_face:)

Thank you again for your work!
Sincerely,
Pandaroux007

**Reply by naveensingh** [reply]
> I wanted to replace the Samsung messaging with `fossify messages'

You won't receive any updates but you should be able to use this older version here: https://github.com/FossifyOrg/Messages/releases/tag/1.0.1

> Can the support stop at version 7 of Android, to support almost all of the Android phones currently in operation?

Sadly no, Android 7 had to go. If it's any consolation, I won't be removing support for Android 8 anytime soon. I would suggest you to try a custom ROM (if you are into customization).

**Reply by TecnikOfficial** [reply]
@pandaroux007 check out these alternative SMS apps for Android 7:
1. **Quik**  
   - **Link**: [Quik on GitHub](https://github.com/octoshrimpy/quik)  
   - **Description**: Another open-source SMS app that focuses on providing a quick and efficient messaging experience.
2. **ConnectYou**  
   - **Link**: [ConnectYou on GitHub](https://github.com/you-apps/ConnectYou)  
   - **Description**: This open-source SMS app works on Android 5 and up. It's privacy-focused, has no ads, and is great for those who care about their privacy.
3. ~~**Simple SMS Messenger**~~  
   - ~~**Link**: [Simple SMS Messenger on GitHub](https://github.com/SimpleMobileTools/Simple-SMS-Messenger)**~~  
   - ~~**Description**: A straightforward SMS app that emphasizes simplicity and is also open-source and ad-free.**~~
   
I wouldn’t recommend installing a custom ROM just to run one app, as custom ROMs come with their own issues and can be risky for beginners. 
Check them out if you want!

**Reply by naveensingh** [reply]
> Link: [Simple SMS Messenger on GitHub](https://github.com/SimpleMobileTools/Simple-SMS-Messenger)

_(I never had anything against the sale and I didn't want to say it but since you linked the app, I think I should)_

The SimpleMobileTools is _not_ the place to go if you want _privacy_ focused apps because the organization on GitHub is also owned by ZippoApps. Tibor doesn't "control" it anymore and the releases there **_can be_** updated silently.

_Edit: clarity._

**Reply by TecnikOfficial** [reply]
yes I know the Simple Mobile Tools GitHub is different from the Play Store release. I thought all FOSSify apps were based on Simple Tools apps since the UI is the same. Ok I updated my original comment and i only  linked the Simple Tools SMS app because you were one of the contributors there. I didn’t realize it was sold as The Simple Tools repo still says it has open-source, ad-free, and privacy-focused apps, but I’m not sure about the Play Store apps.

**Reply by pandaroux007** [reply]
Hello and thank you for these answers (speed that I had never seen on GitHub 👍️)
I intended to use a custom rom but @tecnikofficial is right by saying that it brings other problems and that it is complicated and risky when you are a beginner (like me 🙂) ...

Thank you very much for this list! I am aware that there are other open-source Android messaging applications, but to my knowledge of `SimpleMobileTools` is no longer maintained and really Fossify applications are, in my opinion, the best (tested on another device and I'm not disappointed, everything is perfect!)

I find it a shame not to bear Android 7, but if that is the wish of @Naveensingh and other contributors, let it be so. Your work remains great 👍️

(However, although resigned not to use fossify messages on my samsung on Android 7, I remain a curious developer 🙂 I would like to know what makes the support of Android 7 so complicated/impossible - just to know)

Thanks in advance!
Sincerely,
Pandaroux007

**Reply by TecnikOfficial** [reply]
> (However, although resigned not to use fossify messages on my samsung on Android 7, I remain a curious developer 🙂 I would like to know what makes the support of Android 7 so complicated/impossible - just to know)

Support for older Android versions is not complicated or impossible; it just takes more time. It's entirely up to the developer, as FOSSify is a big project that also manages other apps. The developer may have reduced their workload by discontinuing support for older versions.

**Reply by pandaroux007** [reply]
@TecnikOfficial & @naveensingh Hello and thank you again for all these answers!
> Support for older Android versions is not complicated or impossible; it just takes more time.
> [...]
> The developer may have reduced their workload by discontinuing support for older versions.

Yes, certainly. However, can you not ask the Fossify community and/or contributors to Github to test and compile versions to Android 7? There are certainly enough people in the community to deal with them, leaving you time to develop and thus reduce the workload? Or better still, automate this task - it is most certainly possible, even with the tools offered by Github, right ?

I looked at Android Studio, if you select the Nougat version support (7.0, API 24), you support approximately 97.4% of the devices, against 95.4% for the Oreo version (8.0, API 26) and 93.9% for the Oreo version 8.**1** (API 27). Certainly it is not significant but it still represents people!

This is in no way an obligation, I only get information :+1:
Sincerely,
Pandaroux007

(PS: Sorry again for my English, I am French and I use Google Translation to write: woozy_face :)

**Comment by jankusanagi** [comment]
Here's hoping for a long time support of Android 8! 😅
